test/Microsoft.Azure.Devices.Edge.Test.Common/certs/CertificateAuthority.cs

```diff
@@ -21,6 +25,28 @@ public static async Task<CertificateAuthority> CreateAsync(string deviceId, Root
             return new CertificateAuthority(certs, scriptPath);
         }
 
+        public static async Task<CertificateAuthority> CreateDpsX509CaAsync(string deviceId, RootCaKeys rootCa, string scriptPath, CancellationToken token)
+        {
+            if (!File.Exists(Path.Combine(scriptPath, "certs", "azure-iot-test-only.root.ca.cert.pem")))
```

Comment:
**It's unfortunate that we'd have to reference a hard-coded certificate path in this class. Previously all hard-coded paths were confined to the common OsPlatform class** 
This cert path is hard-coded in `certGen.sh` and `ca-certs.ps1`. The cleanest way to reduce the duplication of this hard code would be to create system environment variables that share between the script and the e2e framework. I currently do not want to make the change right now. 
 - - - - - - - - - - - 
**Does OsPlatform.Current.InstallRootCertificateAsync() fail otherwise?**
The `OsPlatform.Current.InstallRootCertificateAsync()` does NOT fail. However every time it runs a test requiring root cert, it would overwrite and existing one.  In a customer scenario, the root cert should only be installed once anyway, hence I decide to leave the root cert alone if it was installed. In the old E2E scripts only installs the root cert once per device and re-use them across all the Dps test as well.